jetstream_pt/layers.py

```diff
@@ -156,10 +155,7 @@ def __call__(self, xq, xk, xv, mask, cache):
     with jax.named_scope("attn_mat1"):
       ## Attention start
       # scores = torch.einsum(jnp.einsum, "ijkl,ikml->ikjm", xq, keys) / math.sqrt(self.head_dim)
-      scores = torch_xla2.extra.call_jax(
-          jnp.einsum, "ikjl,ikml->ikjm", xq, keys
-      ) / math.sqrt(head_dim)
-      self.env.apply_sharding(scores, axis=1)
```

Comment:
Is matrix transpose HLO issues been fixed?

Reply:
yes

Reply:
Can you share the PR for the fix?

Reply:
And also update b/329899712?

Reply:
https://github.com/pytorch/xla/pull/7009

Reply:
Thank you for the fix! Not directly related, but I see you specially treated einsum to fix the issue. Please also fix b/329899713 for matmul. But the conversation in this PR can be closed now.